src/libraries/System.Text.Json/src/System/Text/Json/Node/JsonObject.IDictionary.cs

```diff
@@ -40,34 +37,23 @@ public void Add(string propertyName, JsonNode? value)
         /// <param name="property">
         ///   The KeyValuePair structure representing the property name and value to add to the <see cref="JsonObject"/>.
         /// </param>
+        /// <exception cref="ArgumentException">
+        ///   An element with the same property name already exists in the <see cref="JsonObject"/>.
+        /// </exception>
         /// <exception cref="ArgumentNullException">
         ///   The property name of <paramref name="property"/> is <see langword="null"/>.
         /// </exception>
         public void Add(KeyValuePair<string, JsonNode?> property)
         {
-            if (property.Key == null)
-            {
-                ThrowHelper.ThrowArgumentNullException_ValueCannotBeNull("propertyName");
-            }
-
-            JsonNode? value = property.Value;
-            value?.AssignParent(this);
-            Dictionary.Add(property);
-            _lastKey = property.Key;
-            _lastValue = value;
+            Add(property.Key, property.Value);
         }
 
         /// <summary>
         ///   Removes all elements from the <see cref="JsonObject"/>.
         /// </summary>
         public void Clear()
         {
-            foreach (JsonNode? node in Dictionary.Values)
-            {
-                DetachParent(node);
-            }
-
-            Dictionary.Clear();
+            ClearNodes();
```

Comment:
nit: could you inline implementation since this is the only caller?

Reply:
The reason for the separation was to keep the list-based code separate from the IDictionary-based code. i.e. the `JsonObject.List.cs` contained the _list and _dictionary variables and corresponding code. However, inlining that method and in general combining the two files makes sense too since that abstraction isn't necessary.